tools/packaging/static-build/cache_components.sh

```diff
@@ -74,10 +74,43 @@ cache_virtiofsd_artifacts() {
 	create_cache_asset "${virtiofsd_tarball_name}" "${current_virtiofsd_version}" "${current_virtiofsd_image}"
 }
 
+cache_rootfs_artifacts() {
+	# We need to remove `-dirty` from teh osbuilder_last_commit as the rootfs artefacts are generated on that folder
+	local osbuilder_last_commit="$(echo $(get_last_modification "${repo_root_dir}/tools/osbuilder") | sed s/-dirty//)"
+	local guest_image_last_commit="$(get_last_modification "${repo_root_dir}/tools/packaging/guest-image")"
+	local agent_last_commit="$(get_last_modification "${repo_root_dir}/src/agent")"
+	local libs_last_commit="$(get_last_modification "${repo_root_dir}/src/libs")"
+	local attestation_agent_version="$(get_from_kata_deps "externals.attestation-agent.version")"
+	local gperf_version="$(get_from_kata_deps "externals.gperf.version")"
+	local libseccomp_version="$(get_from_kata_deps "externals.libseccomp.version")"
+	local pause_version="$(get_from_kata_deps "externals.pause.version")"
+	local skopeo_version="$(get_from_kata_deps "externals.skopeo.branch")"
+	local umoci_version="$(get_from_kata_deps "externals.umoci.tag")"
+	local rust_version="$(get_from_kata_deps "languages.rust.meta.newest-version")"
+	local rootfs_tarball_name="kata-static-cc-rootfs-image.tar.xz"
+	local aa_kbc="offline_fs_kbc"
+	local image_type="image"
+	local root_hash_vanilla="${repo_root_dir}/tools/osbuilder/root_hash_vanilla.txt"
+	local root_hash_tdx=""
+	if [ -n "${TEE}" ]; then
+		if [ "${TEE}" == "tdx" ]; then
```

Comment:
I often get this wrong and struggle to remember, but isn't the 'guide' to use `=` with the 'test'(`[`) command and `==` with `[[` for pattern matching?
As long as `==` works then no need to change it but I've been burned by using the wrong combination before!

Reply:
Hmm.  `[ ${a} = ${b} ]` is POSIX, while `[ ${a} == ${b} ]` is bash specific.
I need to take a look at the scripts but I think we're using the latter in the majority of the places.

Both works as part of our CI, at least.